### ISSUE TITLE
Issues about training from scratch

### ISSUE THREAD

**Issue by lewandofskee** [PKU-YuanGroup/UniWorld-V1#30]
Thank you for your excellent open source work, but I encountered such a problem when reproducing your code. Can you help me find out what the reason is?
In order to quickly verify the experiment, this training only used the viton_hd-23k and shop_product-23k datasets for stage 1 and stage 2 training.
Stage 1 is trained for a total of 500,000 steps. Its training script is:
```
training_config:
  wandb_project: "uniworld"
  wandb_name: "flux_qwen2p5vl_7b_vlm_stage1_512_adamw"
  seed: 203
  output_dir: ./checkpoints/flux_qwen2p5vl_7b_vlm_stage1_512_adamw
  logging_dir: ./logs/flux_qwen2p5vl_7b_vlm_stage1_512_adamw
  max_train_steps: 500000
  optimizer: "adamw"
  learning_rate: 1e-3
  adam_beta1: 0.9
  adam_beta2: 0.99
  adam_epsilon: 1e-8
  adam_weight_decay: 0.0
  mixed_precision: "bf16"
  report_to: "wandb"
  gradient_checkpointing: false
  num_train_epochs: 1
  lr_scheduler: "constant"
  lr_warmup_steps: 0
  lr_num_cycles: 1
  lr_power: 1.0
  resume_from_checkpoint: null
  validation_steps: 1000
  checkpointing_steps: 2000
  drop_t5_rate: 1.0
  discrete_timestep: false
  mask_weight_type: 'log'
  gradient_accumulation_steps: 1
```
The training loss for Stage 1 is:
![Image](https://github.com/user-attachments/assets/9f254511-cc6b-46ad-838c-ec64f158fb71)
However, the visualization results of vlm -> try on in different steps of wandb in stage 1 are rather strange.
However, using the last ckpt for cli testing found that the result was normal, so I took the mlp of stage1 for stage2 training.
The training script for Stage 2 is:
```
training_config:
  wandb_project: "uniworld"
  wandb_name: "flux_qwen2p5vl_7b_vlm_stage2_512_adamw"
  seed: 203
  output_dir: ./checkpoints/flux_qwen2p5vl_7b_vlm_stage2_512_adamw
  logging_dir: ./logs/flux_qwen2p5vl_7b_vlm_stage2_512_adamw
  max_train_steps: 500000
  optimizer: "adamw"
  learning_rate: 1e-3
  adam_beta1: 0.9
  adam_beta2: 0.99
  adam_epsilon: 1e-8
  adam_weight_decay: 0.0
  mixed_precision: "bf16"
  report_to: "wandb"
  gradient_checkpointing: true
  num_train_epochs: 1
  lr_scheduler: "constant"
  lr_warmup_steps: 0
  lr_num_cycles: 1
  lr_power: 1.0
  resume_from_checkpoint: null
  validation_steps: 1000
  checkpointing_steps: 2000
  drop_t5_rate: 1.0
  discrete_timestep: false
  mask_weight_type: 'log'
  gradient_accumulation_steps: 1
```
After training for 140,000 steps, the training loss is：
![Image](https://github.com/user-attachments/assets/5c5e9f71-b8cb-46ae-ab98-53351e115a26)
However, the visualization results in wandb show that there is a problem with the training
![Image](https://github.com/user-attachments/assets/d4610e9f-45a8-46a5-88b4-d9fd36aae500)

I tested it on the shop_product-23k dataset, where stage 1 used a ckpt of 500,000 steps and stage 2 used a ckpt of 140,000 steps. The results are as follows:
![Image](https://github.com/user-attachments/assets/18d60d4d-9590-4189-a31e-a49c96e320a3)

Do you have any idea about this issue?

**Comment by LinB203** [comment]
Sorry, the learning rate for stage 2 is too high. We fixed [it](https://github.com/PKU-YuanGroup/UniWorld-V1/blob/main/scripts/denoiser/flux_qwen2p5vl_7b_vlm_stage2_512.yaml#L9) and recommend 1e-6~1e-5.

**Comment by lewandofskee** [comment]
Thank you for your quick reply. I will change the learning rate of stage 2 and retrain. In addition, I am not sure if there is a problem with the training of stage 1, because the visualization in wandb is a bit strange, but the cli inference is indeed normal.

**Comment by LinB203** [comment]
> Thank you for your quick reply. I will change the learning rate of stage 2 and retrain. In addition, I am not sure if there is a problem with the training of stage 1, because the visualization in wandb is a bit strange, but the cli inference is indeed normal.

I will check the validation function.

**Comment by lewandofskee** [comment]
After modifying the stage 2 learning rate, I have trained for over 80k steps and the generated images are normal. Thanks.
